lib/main_common.pm

```diff
@@ -2401,116 +2401,186 @@ sub load_virtualization_tests {
+            loadtest "installation/scc_registration";
+            loadtest "installation/addon_products_sle";
+            loadtest "installation/system_role";
+            loadtest "installation/partitioning";
+            loadtest "installation/partitioning_firstdisk";
+            loadtest "installation/partitioning_finish";
+            loadtest "installation/installer_timezone";
+            loadtest "installation/user_settings";
+            loadtest "installation/user_settings_root";
+            loadtest "installation/resolve_dependency_issues";
+            loadtest "installation/installation_overview";
+            loadtest "installation/disable_grub_timeout";
+            loadtest "installation/start_install";
+            loadtest "installation/await_install";
+            loadtest "installation/reboot_after_installation";
+        }
     }
 
-    loadtest "virt_autotest/login_console";
-
-    if (check_var('VIRT_PART', 'install')) {
-        loadtest 'virtualization/universal/prepare_guests';    # Prepare libvirt and install guests
-        loadtest 'virtualization/universal/ssh_hypervisor_init';    # Configure SSH for hypervisor
-        loadtest 'virtualization/universal/waitfor_guests';    # Wait for guests to be installed
-
-        loadtest 'virtualization/universal/ssh_guests_init';    # Fetch SSH key from guests and connect
-        loadtest 'virtualization/universal/register_guests';    # Register guests against the SMT server
-        loadtest 'virtualization/universal/upgrade_guests';    # Upgrade all guests
-        loadtest 'virtualization/universal/patch_guests';    # Apply patches to all compatible guests
-        loadtest 'virtualization/universal/patch_and_reboot';    # Apply updates and reboot
-
+    # QAM VM installation
+    unless (get_var('DISABLE_VM_INSTALL')) {
         loadtest "virt_autotest/login_console";
-    }
-
-    loadtest "virtualization/universal/list_guests" unless ($windows);    # List all guests and ensure they are running
-
-    if (check_var('VIRT_PART', 'install')) {
-        loadtest "virtualization/universal/kernel";    # Virtualization kernel functions
-    }
-
-    if (check_var('VIRT_PART', 'virtmanager')) {
-        loadtest 'virtualization/universal/virtmanager_init';    # Connect to the Xen hypervisor using virt-manager
-        loadtest 'virtualization/universal/virtmanager_offon';    # Turn all VMs off and then on again
-
-        if (is_sle('12-SP3+')) {
-            loadtest 'virtualization/universal/virtmanager_add_devices';    # Add some aditional HV to all VMs
-            loadtest 'virtualization/universal/virtmanager_rm_devices';    # Remove the aditional HV from all VMs
+        if (check_var('INSTALL_GUEST', 1)) {
+            # Prepare and wait for guests
+            loadtest "virtualization/universal/prepare_guests";
+            loadtest "virtualization/universal/waitfor_guests";
+        }
+        if (check_var('PATCH_WITH_ZYPPER', 1)) {
```

Comment:
Although in ticket requirement, it mentions adding one switch for `qam-kvm-install` function, I guess for real debug, there will be such need: skip installing guest or skip updating guest, or both, or completely. Will it? 
- If yes, suggest to update the name 'INSTALL_GUEST' to switch style `DISABLE_XX`. 
- Otherwise, to me, 'INSTALL_GUEST' is not very necessary. Keeping the top switch 'DISABLE_VM_INSTALL' is enough.

HDYT?

Reply:
I agree with you to drop such switch 'INSTALL_GUEST', it comes from old yaml schedule logic.